TuneBot/Modules/TrackModule.cs

```diff
@@ -2,331 +2,282 @@
 using Discord.Commands;
 using Google.Apis.YouTube.v3;
 using IF.Lastfm.Core.Api;
-using Microsoft.Data.Sqlite;
 using Microsoft.Extensions.Configuration;
 using SpotifyAPI.Web;
 using System;
 using System.Linq;
 using System.Threading.Tasks;
+using Discord;
+using Discord.Interactions;
+using TuneBot.Services.DataAccess;
 
 namespace TuneBot.Modules
 {
-    public class TrackModule : ModuleBase<SocketCommandContext>
+    public class TrackModule : InteractionModuleBase<SocketInteractionContext>
     {
-        private readonly YouTubeService youtube;
-        private readonly IConfiguration config;
+        private readonly YouTubeService _youtubeService;
+        private readonly IConfiguration _configuration;
+        private readonly DbService _dbService;
 
-        public TrackModule(YouTubeService youtube, IConfiguration config)
+        public TrackModule(YouTubeService youtubeService, IConfiguration configuration, DbService dbService)
         {
-            this.youtube = youtube;
-            this.config = config;
+            _youtubeService = youtubeService;
+            _configuration = configuration;
+            _dbService = dbService;
         }
 
-        [Command("track")]
-        [Summary("Gets a track from Spotify and YouTube.")]
+        [SlashCommand("track", "Gets a track from Spotify and YouTube.")]
         public async Task GetTrack([Remainder] string details)
         {
             using var typing = Context.Channel.EnterTypingState();
 
-            var spotifyResultTask = SearchSpotify(details);
-            var youTubeResultTask = SearchYouTube(details);
-            await Task.WhenAll(spotifyResultTask, youTubeResultTask);
-           
-            await ReplyAsync(spotifyResultTask.Result ?? "No Spotify match found.");
-            await ReplyAsync("\n" + youTubeResultTask.Result ?? "No YouTube match found");
+            var spotifyResultTask = SearchSpotifyAsync(details);
+            var youTubeResultTask = SearchYouTubeAndGetLinkAsync(details);
+
+            var (spotifyResult, youtubeResult) = (await spotifyResultTask, await youTubeResultTask);
+
+            await RespondAsync(spotifyResult is null ? "No Spotify result found" : spotifyResult.Uri);
```

Comment:
This responds with spotify's internal URI of a track e.g. "spotify:track:0P1jQfRErBjgOSTbVKtCW7". Looking at the old code it to get the public URL it's `track.ExternalUrls["spotify"]`